lib/include/openamp/remoteproc.h

```diff
@@ -395,6 +395,13 @@ struct remoteproc {
  *        memory may not be off.
  * @shutdown: shutdown the remoteproc and release its resources.
  * @notify: notify the remote
+ * @get_mem: get remoteproc memory I/O region by either name, virtual address,
+ *           physical address or device address.
+ *           Note:
+ *           1.remoteproc_get_io_with_xxx try the internal list added by
+ *             remoteproc_add_mem first and then get_mem.
+ *           2.get_mem could return either internal managed instance or
+ *             the 'buf' argument as needed.
```

Comment:
No clear that the buff is optional... proposal inspirated from https://en.cppreference.com/w/c/chrono/gmtime:
`if "buf" parameter is not specified the result is stored in static remoteproc_mem structure and a pointer to that static storage is returned. Else uses provided storage "buf" for the result and return the associated pointer.`

Reply:
buf isn't optional here. the caller always pass buf, but the implemenation could select wether to use this buffer or return an internal instance.